Add age bar

moos opened the pull request:
Basically inspired by http://www.bertrandplanes.com/pages/LifeClock3.php.

Also:
- Fixes center text alignment on first draw
- Ensures font is loaded
- Use more accurate seconds update using setTimeout
- Added ability to scale all bars

moos committed:
Add age bar

moos committed:
Add age bar

moos committed:
Only update seconds hand on the second

moos committed:
Fix tabs

moos committed:
Add birth month; revert to original defaults

moos committed:
Add birth month; revert to original defaults

moos committed:
Add birth month; revert to original defaults

eouia commented:
Interesting. 
Can you add a more detailed explanation on README?
The other issue would be, you've added a hand for life age, but is there risk of conflict existing configuration with this new version?

eouia committed:
Merge pull request #3 from moos/perf

Only update seconds hand on the second

moos committed:
Merge branch 'age' of https://github.com/moos/MMM-OClock into age

moos committed:
Use 'age' hand to not conflict with 'year'.  Update README

moos commented on MMM-OClock.js:
```diff
@@ -31,8 +32,15 @@ Module.register("MMM-OClock", {
     colorTypeRadiation: ["#333333", "red"], //Don't use #pattern or colorName.
     colorTypeTransform: ["blue", "red"],
     colorTypeHSV: 0.25, //hsv circle start color : 0~1
+    secondsUpdateInterval: 1000,  // msec
```
Forgot to add this in previous PR.

moos commented on MMM-OClock.js:
```diff
@@ -31,8 +32,15 @@ Module.register("MMM-OClock", {
+    linearLife: false,  // set to true to plot life linearly not logarithmically
 
     handConversionMap: {
+      "age": "n/a",
```
Use separate name so not to conflict with existing "year".

moos committed:
Add code comments

moos commented:
I merged from master before realizing you had also merged the other PR.  Also addressed your comments.  Now using "age" so it doesn't conflict with existing config files.

Here's a screenshot (46, not my true age!):

![oclock-age](https://user-images.githubusercontent.com/233047/71759525-0d6b5780-2e63-11ea-8f3a-ceb59fb09ebb.png)

moos committed:
Add 'scale' config param to scale all bar widths

moos committed:
More accurate timing update using setTimeout instead of setInterval. Fix center text alignment

moos committed:
Ensure font is loaded before first draw

moos commented:
OK -- think I'm done.  Hopefully you'll find this useful.

I think the age bar needs some decoration to stand-out from the others.  Maybe an exercise for someone else.

moos committed:
Adding canvasStyle config param

moos commented:
Oops, I lied.  I use [delightedCrow/WallberryTheme](https://github.com/delightedCrow/WallberryThem) and the OClock looks too contrast-y.  So adding ability to add CSS via config.
